lib/mutant/mutator/node/regexp.rb

```diff
@@ -0,0 +1,104 @@
+module Mutant
+  class Mutator
+    class Node
+
+      # Generic mutator
+      module RegularExpression
+        class Parser
+          def self.parse(source)
+            Node.new(Regexp::Parser.parse(source, 'ruby/2.3'))
+          end
+
+          class Node < DelegateClass(Regexp::Expression::Base)
```

Comment:
If you are ok with it I think this might might be worth punting. Right now, if you only swap out the `#expressions` attribute on a `Regexp::Expression` instance, then you get unparsing for free. This might be much more complex if everything is mapped to a `Parser::AST`. WDYT?